river/src/main/java/org/jboss/marshalling/river/RiverUnmarshaller.java

```diff
@@ -207,6 +207,8 @@ Object doReadNestedObject(final boolean unshared, final String enclosingClassNam
         } catch (RuntimeException e) {
             TraceInformation.addIncompleteObjectInformation(e, enclosingClassName);
             throw e;
+        } catch (StackOverflowError e) {
+            throw new RuntimeException(e);
```

Comment:
Should we be calling `TraceInformation.addIncompleteObjectInformation` here too? I suppose it would be added by the next stack frame up?